### DISCUSSION TITLE
Accessing prev/today/next functionality from my own navbar?

### DISCUSSION THREAD

**Discussion by hpaves** [vkurko/calendar#252] [Q&A] [answered]
I would like to mount your fantastic piece of software to a basic Svelte project, and build a navbar to it. However, your calendar has an inbuilt programmatically generated toolbar. I haven't been able to figure out how to include prev/today/next buttons in my own navbar, as well as other functionalities, such as the view changing buttons, and the "title" (the span of time in the current view). Could you please give me some pointers on how this could be accomplished?

**Comment by vkurko** [comment, marked as answer]
To navigate the calendar programmatically, you should update the [date](https://github.com/vkurko/calendar?tab=readme-ov-file#date) option using [setOption](https://github.com/vkurko/calendar?tab=readme-ov-file#setoption-name-value-), i.e. `calendar.setOption('date', newDate)`. Accordingly, to switch the [view](https://github.com/vkurko/calendar?tab=readme-ov-file#view) you need to use `setOption('view', newViewName)`. And the current title value can be obtained from the [View](https://github.com/vkurko/calendar?tab=readme-ov-file#view-object) object using [getView](https://github.com/vkurko/calendar?tab=readme-ov-file#getview).

**Reply by hpaves** [reply]
This worked great. Thanks!
